### PR TITLE
feat(attention): add a Triton dense backend for gfx1250

### PR DESCRIPTION
**Pull request by JohnQinAMD**
Add dense backend runs on gfx1250. FlyDSL and HipKittens are built for other targets, and the aiter path reaches CK, whose fmha kernels are CDNA-only: the forward succeeds and the backward then rejects the call at runtime with "invalid argument for fmha_bwd", so the failure lands mid training step rather than at dispatch.

attention_triton_impl.py already carries a portable Triton forward and backward -- no CK, no aiter, no arch-specific extension -- but nothing reached them. This registers them as a backend:

- dense_forward/dense_backward adapt the impls for bf16/fp16. Their signatures carry the fp8 plumbing (descales, p_scale), which is identity for a non-fp8 call, so fill it in once here instead of at both call sites. dq/dk/dv go in as None: they are returns of the custom op, and passing preallocated tensors trips torch's aliasing check.
- DenseAttnFwdTritonBackend declares what the kernel actually asserts -- bshd, no sliding window, no bias/alibi/sink/dropout -- and refuses anything else so the dispatcher can pick a backend that handles it.
- can_handle is gated on is_gfx1250 for now. The kernel is portable and may well win elsewhere, but it has only been validated here, and an unvalidated backend joining the autotune race could be picked silently on hardware nobody checked. Widening it later is one line (cf. is_gfx950 on HipKittens). Until then nothing changes on any other target.

Measured on gfx1250,  Numerics against SDPA: out 58.1 dB, dq 58.1, dk 56.9, dv 79.7.

Not covered: varlen (the Triton forward takes no cu_seqlens, so that needs kernel work rather than registration) and the USP path.

# Description

Please include a brief summary of the changes, relevant motivation and context.

Fixes # (issue)

## Type of change

- [ ] Documentation change (change only to the documentation, either a fix or a new content)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Infra/Build change
- [ ] Code refactoring

## Changes

Please list the changes introduced in this PR:

- Change A
- Change B

# Checklist:

- [ ] The functionality is complete
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by RuibinCheung — DISMISSED** [review]
LGTM if CI is green.

File: primus_turbo/pytorch/kernels/attention/attention_impl.py [hunk 1]
```diff
@@ -162,8 +165,6 @@ def can_handle(
         k=None,
         v=None,
         dropout_p=0.0,
```
[the remark this answers is not in the document]
Comment (JohnQinAMD) [reply primus_turbo/pytorch/kernels/attention/attention_impl.py:206]: fixed

**Review by RuibinCheung — APPROVED** [review]
LGTM
